aea/configurations/base.py

```diff
@@ -612,5 +630,7 @@ def from_json(cls, obj: Dict):
         # set default configuration
         default_connection_name = obj.get("default_connection", None)
         agent_config.default_connection = default_connection_name
+        default_ledger_id = obj.get("default_ledger", None)
+        agent_config.default_ledger = default_ledger_id
```

Comment:
The `ledger_id` setter above accepts only `str`, but `default_ledger_id` can be `None`

Reply:
then we should change it for the default connection too